models/workitemtype_repository.go

```diff
@@ -11,6 +11,8 @@ import (
 	"github.com/jinzhu/gorm"
 )
 
+var cache = NewWorkItemTypeCache()
```

Comment:
There are a few places where WorkItemTypeRepository is created.
- Application
- WorkItemRepository (created in Application and remoteworkitem.convert())
- Search Repository
- in main() for migration.PopulateCommonTypes() (which is not a big deal since we need to use a different cache or clear the global one there anyway)

So if we want to pass a wit cache when creating WIT Repository we could keep it in the GormBase and create a new temporary instance for the other consumers from the list above.

I would rather keep the cache as a global unexported variable in WorkItemTypeRepository so we can benefit the really global wit cache plus the code would be simpler. WDYT?